content/en/blog/_posts/2024-08-19-pod-failure-policy-ga.md

```diff
@@ -0,0 +1,204 @@
+[graceful node shutdown](/docs/concepts/cluster-administration/node-shutdown/#graceful-node-shutdown).
+
+The policy allows you to distinguish between retriable and non-retriable Pod
+failures based on container exit codes or Pod conditions in a failed Pod.
+
+## How it works
+
+You specify a Pod failure policy in the Job specification, represented as a list
+of rules.
+
+For each rule you define _match requirements_ based on one of the following properties:
+
+- Container exit codes: the `onExitCodes` property.
+- Pod conditions: the `onPodConditions` property.
+
+Additionally, for each rule, you specify one of the following actions to take
+when a Pod matches the rule:
+- `Ignore`: Do not count the failure towards the `backoffLimit` or `backoffLimitPerIndex`.
+- `FailJob`: Fail the entire Job and terminate all running Pods.
+- `FailIndex`: Fail the index corresponding to the failed Pod
+  [Backoff limit per index](/docs/concepts/workloads/controllers/job/#backoff-limit-per-index),
+- `Count`: Count the failure towards the `backoffLimit` or `backoffLimitPerIndex`.
+  This is the default behavior.
+
+When Pod failures occur in a running Job, Kubernetes matches the
+failed Pod status against the list of Pod failure policy rules, in the specified
+order, and takes the corresponding actions for the first matched rule.
+
+### Kubernetes-initiated Pod disruptions
+
+To allow matching Pod failure policy rules against failures caused by
+disruptions initiated by Kubernetes, this feature introduces the `DisruptionTarget`
+Pod condition.
+
+This condition is added by Kubernetes when a Pod fails because of a retriable
+[disruption scenario](/docs/concepts/workloads/pods/disruptions/#pod-disruption-conditions).
+The `DisruptionTarget` condition contains one of the following reasons that corresponds to
+these disruption scenarios:
+
+- `PreemptionByKubeScheduler`: [Preemption](/docs/concepts/scheduling-eviction/pod-priority-preemption)
+   by `kube-scheduler` to accommodate a new Pod that has a higher priority.
+- `DeletionByTaintManager` - the Pod is due to be deleted by
+   `kube-controller-manager` due to a `NoExecute` [taint](/docs/concepts/scheduling-eviction/taint-and-toleration/)
+   that the Pod doesn't tolerate.
+- `EvictionByEvictionAPI` - the Pod has been marked for
+   [eviction using the Kubernetes API](/docs/concepts/scheduling-eviction/api-eviction/).
+- `DeletionByPodGC` - the Pod is bound to a node that no longer exists, and is due to
+   be deleted by [Pod garbage collection](/docs/concepts/workloads/pods/pod-lifecycle/#pod-garbage-collection).
+- `TerminationByKubelet` - the Pod was terminated by
+  [graceful node shutdown](/docs/concepts/cluster-administration/node-shutdown/#graceful-node-shutdown),
+  [node pressure eviction](/docs/concepts/scheduling-eviction/node-pressure-eviction/)
+  or preemption for [system critical pods](/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/).
+
+### Examples
```

Comment:
I tentaively split the single example into 3 smaller in this commit: https://github.com/kubernetes/website/pull/46896/commits/227acb546c77615fac490ba5718f0bb6a658edbf. 

Motivation:
- to also describe FailIndex for backoff limit per index
- to make smaller, more focused snippets, which are easier to describe

WDYT @tenzen-y @atiratree @shannonxtreme ?

Reply:
I say no. The task page has that already, split into three, only thing that's missing is the index one. Blogs go out of date fast, docs don't, so recommend that you save details like this for the docs and rely on them. If content from your commit isn't already in https://kubernetes.io/docs/tasks/job/pod-failure-policy/ put it there at some point

Reply:
Given the feedback I reverted the last commit, but I'm happy to reconsider some parts of it if other find it useful - the link to the commit still works.

Reply:
TBH, I don't have a strong view of this example, but @shannonxtreme opinion sounds reasonable.
In general, blog articles are easy to outdated.

Reply:
/lgtm

Reply:
sgtm, I think we add extending the tutorial for backoff limit per index + pod failure policy, as a graduation criteria for backoff limit per index to stable. WDYT?

Reply:
Does it mean we want to add some more examples for the back off limit per index + pod failure policy to the documentation, not blog?

It sounds attractive, but maybe we can revisit this discussion when the backoff limit per index feature graduate to stable.

Reply:
> Does it mean we want to add some more examples for the back off limit per index + pod failure policy to the documentation, not blog?

Yeah, this is what I was considering

> It sounds attractive, but maybe we can revisit this discussion when the backoff limit per index feature graduate to stable.

sgtm

Reply:
Sound good to me.